# Step 01. The soft-thresholding power

Weighted gene co-expression network analysis (WGCNA) raises every protein-protein correlation to a
power, so weak correlations collapse toward zero and strong ones survive. Everything downstream
depends on this one number.

**This repository takes the source paper's value.** Leung GHD et al., *Mapping the plasma proteomic
architecture of systemic lupus erythematosus*, JCI Insight 2026 (doi:10.1172/jci.insight.206938),
Supplemental Methods section 1: "An unsigned network was constructed, and a soft thresholding power
of 6 was selected based on the standard criteria of approximate scale-free topology and preservation
of mean network connectivity."

So the network is **unsigned** and the power is **6**, not because a criterion applied here selected
them but because they are the published values and the object of this analysis is to see whether the
published result reproduces. The scan below is run anyway, and reported, because the reader is
entitled to know what our own data would have chosen.

In [1]:
source("../src/paths.R")
suppressMessages(library(WGCNA))
options(stringsAsFactors = FALSE); enableWGCNAThreads(4)
COHORT <- "A"
X <- read.csv(coh("R_cohort-%s_log2_combat.csv", COHORT), row.names = 1, check.names = FALSE)
sprintf("%d patients x %d proteins", nrow(X), ncol(X))

Allowing parallel execution with up to 4 working processes.


[1] "87 patients x 7288 proteins"

In [2]:
sft <- pickSoftThreshold(X, powerVector = 1:12, networkType = "unsigned", verbose = 0)
fi  <- sft$fitIndices[, c("Power", "SFT.R.sq", "slope", "mean.k.")]
print(round(fi, 3), row.names = FALSE)
saveRDS(fi, art("sft_unsigned_%s.rds", COHORT))

   Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
1      1   0.0214 -0.0989          0.751  1630.0  1540.000   3040
2      2   0.9060 -0.7830          0.938   607.0   469.000   1740
3      3   0.9610 -0.9540          0.956   293.0   170.000   1160
4      4   0.9520 -1.0200          0.943   166.0    69.100    841
5      5   0.9360 -1.0700          0.926   104.0    30.600    655
6      6   0.9270 -1.0900          0.923    70.0    14.600    530
7      7   0.9220 -1.1200          0.924    49.7     7.380    439
8      8   0.8930 -1.1500          0.907    36.7     3.980    371
9      9   0.8890 -1.1600          0.913    27.9     2.230    318
10    10   0.8610 -1.2000          0.901    21.7     1.330    275
11    11   0.8580 -1.2000          0.911    17.2     0.840    240
12    12   0.8460 -1.2200          0.914    13.9     0.553    211


 Power SFT.R.sq  slope  mean.k.
     1    0.021 -0.099 1630.087
     2    0.906 -0.783  606.976
     3    0.961 -0.954  292.982
     4    0.952 -1.017  165.584
     5    0.936 -1.065  103.830
     6    0.927 -1.093   69.974
     7    0.922 -1.120   49.678
     8    0.893 -1.149   36.669
     9    0.889 -1.163   27.890
    10    0.861 -1.197   21.722
    11    0.858 -1.204   17.245
    12    0.846 -1.223   13.909


**Explanation**
- `pickSoftThreshold()` fits, at each candidate power, a scale-free topology model to the resulting
  network's connectivity distribution, and reports `SFT.R.sq`, the R² of that fit, together with
  `mean.k.`, the mean connectivity.
- `networkType = "unsigned"` is the paper's choice. An unsigned network uses the **absolute**
  correlation, so a protein and something moving opposite to it can share a module. A signed network
  uses `(1 + r)/2` and joins only proteins moving the same way. The two build different objects, and
  the power that suits one does not suit the other: for a signed network the WGCNA authors publish a
  floor of 12 at n > 40, which is what this notebook used before and which is why the power changed
  from 12 to 6 at the same time as the network type. **Neither number is meaningful without the
  other.**
- `1:12` is a wider range than needed, so the whole curve is visible rather than just the chosen point.

## What the paper's own criterion gives on this sample

In [3]:
POWER <- 6                       # the paper's value; see the note above
lowest <- fi$Power[fi$SFT.R.sq > 0.9]
cat(sprintf("cohort %s, n = %d, unsigned network\n", COHORT, nrow(X)))
cat(sprintf("  lowest power with R2 > 0.9 : %s   <- what the paper's stated criterion gives here\n",
            if (length(lowest)) min(lowest) else NA))
cat(sprintf("  pickSoftThreshold estimate : %s\n", sft$powerEstimate))
cat(sprintf("  the paper's power          : %d, R2 = %.3f, mean connectivity %.1f\n",
            POWER, fi$SFT.R.sq[fi$Power == POWER], fi$mean.k.[fi$Power == POWER]))
cat(sprintf("\nusing POWER = %d, to match the published analysis\n", POWER))

cohort A, n = 87, unsigned network


  lowest power with R2 > 0.9 : 2   <- what the paper's stated criterion gives here


  pickSoftThreshold estimate : 2


  the paper's power          : 6, R2 = 0.927, mean connectivity 70.0



using POWER = 6, to match the published analysis


**Explanation**
- `fi$Power[fi$SFT.R.sq > 0.9]` are the powers that clear the scale-free threshold the paper names.
  `min()` of those is the criterion it states: "the lowest power to achieve a scale-free topology
  (R² > 0.9)".
- `POWER <- 6` is fixed, not derived. The two lines above it exist so the gap between the published
  value and what this sample would have chosen is on the page rather than in a footnote.

**Result.** On cohort A the lowest power clearing R² > 0.9 is **2**, and `pickSoftThreshold`'s own
estimate is also 2. The paper's power of **6** gives R² = **0.927** here, so it clears the threshold
comfortably; it is simply not the *lowest* power that does. On 207 patients the same criterion gave
them 6.

Across the three cohorts, at power 6: **A R² = 0.927, C R² = 0.900, B R² = 0.865.** Cohort B falls
below the threshold the criterion names. That is one more instance of the pattern step 11 documents,
and it is a reason to read B's modules with care rather than a reason to change B's power, which
would make the three cohorts incomparable.

**What this means for the comparison.** Using 6 is a choice to match the published pipeline, not a
claim that 6 is optimal for n = 87. A sample less than half the size gives a flatter scale-free
curve, so the criterion bites at a much lower power. Both facts are reported; neither is resolved by
this notebook.

## Where this was run

Rendered notebooks are committed, so each records the machine, the R and the
package versions that produced its output.


In [4]:
# Provenance. The rendered notebook is committed, so it records where it ran.
# The cell is identical in every notebook and is explained in step 00.
si <- Sys.info()
cat("run on      :", si[["nodename"]], "(", si[["sysname"]], si[["release"]], ")\n")
cat("date        :", format(Sys.time(), "%Y-%m-%d %H:%M %Z"), "\n")
cat("R           :", R.version.string, "|", R.version$platform, "\n")
cat("R comes from:", R.home(), "\n")
cat("packages    :\n")
for (p in c("WGCNA", "ComplexHeatmap", "circlize", "ReactomePA", "reactome.db",
            "org.Hs.eg.db", "sva", "VarSelLCM"))
  cat(sprintf("   %-16s %s\n", p,
              tryCatch(as.character(packageVersion(p)), error = function(e) "not installed")))

run on      : Annes-MacBook-Pro-193.local ( Darwin 27.0.0 )


date        : 2026-09-28 02:34 EDT 


R           : R version 4.4.3 (2025-02-28) | x86_64-apple-darwin13.4.0 


R comes from: /Users/adeslatt/miniforge3/envs/endotypes-proteomics/lib/R 


packages    :


   WGCNA            1.74
   ComplexHeatmap   2.22.0
   circlize         0.4.18
   ReactomePA       1.50.0
   reactome.db      1.89.0
   org.Hs.eg.db     3.20.0
   sva              3.54.0
   VarSelLCM        2.1.3.2
